SlevomatCodingStandard/Sniffs/ControlStructures/AssignmentInConditionSniff.php

```diff
@@ -54,11 +62,36 @@ public function process(File $phpcsFile, $conditionStartPointer): void
 
 	private function processCondition(File $phpcsFile, int $parenthesisOpener, int $parenthesisCloser, string $conditionType): void
 	{
-		$equalsTokenPointer = TokenHelper::findNext($phpcsFile, T_EQUAL, $parenthesisOpener + 1, $parenthesisCloser);
-		if ($equalsTokenPointer === null) {
+		$equalsTokenPointers = TokenHelper::findNextAll($phpcsFile, T_EQUAL, $parenthesisOpener + 1, $parenthesisCloser);
+		$tokens = $phpcsFile->getTokens();
+
+		if (!$this->ignoreAssignmentsInsideFunctionCalls && count($equalsTokenPointers) !== 0) {
+			$this->error($phpcsFile, $conditionType, reset($equalsTokenPointers));
 			return;
 		}
 
+		foreach ($equalsTokenPointers as $equalsTokenPointer) {
+			$parenthesisStarts = array_keys($tokens[$equalsTokenPointer]['nested_parenthesis'] ?? []);
+			$insideParenthesis = max($parenthesisStarts);
+			if ($insideParenthesis === $parenthesisOpener) {
+				$this->error($phpcsFile, $conditionType, $equalsTokenPointer);
+				//this is assignment inside if expression
+				continue;
```

Comment:
The `continue` makes sense only if we report all assignments in one condition. Can you please add test that reports it?

Something like `if ($a = 1 && $b = 1)`